instruqt-tracks/terraform-cloud-aws-v2/02-terraform-cloud-setup/check-workstation

```diff
@@ -19,14 +36,85 @@ cd /root/hashicat-aws
+cd /root/terraform-cloud
+sed -i "s/YOUR_ORGANIZATION/$ORG/g" terraform.tfvars # This was originally looking in terraform.tfvars.example
+sed -i "s/YOUR_WORKSPACE/$WORKSPACE/g" terraform.tfvars
+# mv terraform.tfvars.example terraform.tfvars
+# The above command will get rid of the terraform.tfvars.example file, but when
+# we run the check script more than once, it cannot find that file bc it no longer
+# exists. However, to keep the consistency between code blocks, we kept this the 
+# same as above where we search and replace in the remote_backend.tf file. 
 
 # Store the ORG in /root/.bashrc
 grep $ORG /root/.bashrc || echo "export ORG=\"$ORG\"" >> /root/.bashrc
 
 # Store the WORKSPACE in /root/.bashrc
 grep $WORKSPACE /root/.bashrc || echo "export WORKSPACE=\"$WORKSPACE\"" >> /root/.bashrc
 
-exit 0
+# Do we have a valid token
+if [ -z "${TFC_TOKEN}" ]; then
+    fail-message "Unable to find Terraform Cloud Token, please double-check the \"3- Terraform Cloud Token\" steps"
+    exit 1
+fi
+
+STATUS=$(curl \
+    --header "Authorization: Bearer ${TFC_TOKEN}" \
+    --header "Content-Type: application/vnd.api+json" \
+    --request GET \
+    -w "%{response_code}" \
+    -s \
+    -o /tmp/.out.json \
+    https://app.terraform.io/api/v2/account/details 2>/dev/null)
+
+if [ "${STATUS}" != "200" ]; then
+    echo "Failed to get account details, status ${STATUS}"
+    if [ -f /tmp/.out.json ]; then
+	echo "Output was:"
+	cat /tmp/.out.json
+	echo "End of output"
+    fi
+    fail-message "Terraform Cloud Token is not valid, please double-check the \"3- Terraform Cloud Token\" steps"
+    exit 1
+fi
+
+rm -f /tmp/.out.json
+echo "We have a valid TFC token"
+
+# Is it a trial organization
+STATUS=$(curl \
+    --header "Authorization: Bearer ${TFC_TOKEN}" \
+    --header "Content-Type: application/vnd.api+json" \
+    --request GET \
+    -w "%{response_code}" \
+    -s \
+    -o /tmp/.out.json \
+    https://app.terraform.io/api/v2/organizations/"${ORG}"/entitlement-set 2>/dev/null) # originally TFC_ORG
+
+if [ "${STATUS}" != "200" ]; then
+    echo "Failed to get organization entitlements, status ${STATUS}"
+    if [ -f /tmp/.out.json ]; then
+	echo "Output was:"
+	cat /tmp/.out.json
+	echo "End of output"
+    fi
+    fail-message "Unable to get TFC Organization Entitlements, please go back to the \"2- Terraform Cloud Trial Plan\" steps"
+    exit 1
+fi
+
+TRIAL=$(jq -r '.data.attributes."versioned-policy-set-limit"' /tmp/.out.json 2>/dev/null)
+if [ -z "${TRIAL}" ]; then
+    echo ".data.attributes.versioned-policy-set-limit was empty"
+    fail-message "Unable to determine TFC Trial Status, please go back to the \"2- Terraform Cloud Trial Plan\" steps"
+    exit 1
+fi
+
+if [ "${TRIAL}" != "null" ]; then
+    echo ".data.attributes.versioned-policy-set-limit was '${TRIAL}' not 'true'"
+    fail-message "Your TFC Organization is not a trial one, please go back to the \"2- Terraform Cloud Trial Plan\" step"
+    exit 1
+fi
```

Comment:
I'm not sure this logic is going to give us what we need and may be prone to false positives. I'm currently testing with trial orgs and will report back.

Reply:
@cstella84 did you managed to do this testing? I think pragmatically what we have now is good enough, but I wanted to double-check with you.

Reply:
Ah, in discussion with Jenna, it appears this comment was back when we were checking `versioned-policy-set-limit` — what's in the PR now is completely different, looking for `
TRIAL=$(jq -r '.included[].attributes."identifier"' /tmp/.out.json 2>/dev/null)` to be `trial`